Doc/library/hashlib.rst

```diff
@@ -300,23 +300,16 @@ include a `salt <https://en.wikipedia.org/wiki/Salt_%28cryptography%29>`_.
 
    >>> from hashlib import pbkdf2_hmac
    >>> our_app_iters = 500_000  # Application specific, read above.
-   >>> dk = pbkdf2_hmac('sha256', b'password', b'bad salt'*2, our_app_iters)
+   >>> dk = pbkdf2_hmac('sha256', b'password', b'bad salt' * 2, our_app_iters)
    >>> dk.hex()
    '15530bba69924174860db778f2c6f8104d3aaf9d26241840c8c4a641c8d000a9'
 
-   .. versionadded:: 3.4
-
-   .. note::
+   Function only available when Python is compiled with OpenSSL.
 
-      A fast implementation of *pbkdf2_hmac* is available with OpenSSL.  The
-      Python implementation uses an inline version of :mod:`hmac`. It is about
-      three times slower and doesn't release the GIL.
-
-   .. deprecated:: 3.10
+   .. versionadded:: 3.4
 
-      Slow Python implementation of *pbkdf2_hmac* is deprecated. In the
-      future the function will only be available when Python is compiled
-      with OpenSSL.
+   .. versionchanged:: 3.12
+      Remove the slow pure Python implementation, deprecated in Python 3.10.
```

Comment:
The remark is hard to understand for users that are not aware of the history of the function. I would rephrase it to something like "pbkdf2_hmac is now only available when Python is built with OpenSSL. The slow pure Python implementation has been removed."

Reply:
I copied your text, thanks.